quasar/src/mixins/validate.js

```diff
@@ -52,39 +53,97 @@ export default {
       this.innerError = false
       this.innerErrorMessage = void 0
     },
+    getRule (original) {
+      let rule = {}
+      rule.handler = () => true
+      rule.args = undefined
+      rule.lazy = this.lazyRules
+      rule.disabled = this.disabledRules
+      rule.message = undefined
+      rule.label = this.validationLabel || this.label
 
-    validate (val = this.value) {
-      let msg, error = false
+      if (typeof original === 'object') {
+        rule.args = original.args || rule.args
+        rule.lazy = original.lazy || rule.lazy
+        rule.disabled = original.disabled || rule.disabled
+        rule.message = original.message || rule.message
+        rule.label = original.label || rule.label
+        original = original.handler
+      }
 
-      for (let i = 0; i < this.rules.length; i++) {
-        const rule = this.rules[i]
-        if (typeof rule === 'function') {
-          const res = rule(val)
+      if (typeof original === 'function') {
+        rule.handler = original
+      }
 
-          if (typeof res === 'string') {
-            error = true
-            msg = res
-            break
-          }
+      if (typeof original === 'string') {
+        if (testPattern[original] !== void 0) {
+          rule.handler = testPattern[original]
         }
-        else if (typeof rule === 'string' && testPattern[rule] !== void 0) {
-          if (testPattern[rule](val) !== true) {
-            error = true
-            break
+        else {
+          try {
+            var regex = new RegExp(original)
+            rule.handler = (value) => regex.test(value)
           }
+          finally { }
```

Comment:
??? a finally without a fall-through? using try without a catch?

Reply:
how i would validate if a string is a valid regular expression?